### PR TITLE
added documentation about user-managed deployments

### PR DESCRIPTION
**Pull request by warber**
Signed-off-by: warber <bernd.warmuth@dynatrace.com>

Preview: https://deploy-preview-762--keptn.netlify.app/docs/0.8.x/continuous_delivery/deployment_helm/

### CODE REVIEW

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 1]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
 carts           0/0     0            0            1d   carts        docker.io/keptnexamples/carts:0.11.2   app=carts
 ```
 
+
+## User Managed deployments (experimental)
+
+*Note: This feature is currently marked as experimental. Hence, expect various enhancements and/or changes in
```
Comment (johannes-b) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:81]: You can put a `:warning:` in front.

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 2]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+future releases of Keptn.*
+
+
+If the deployment strategy of a stage in the [shipyard](https://github.com/keptn/examples/blob/0.8.0/onboarding-carts/shipyard.yaml)
+is configured as `deploymentstrategy: user_managed`, the provided helm chart will be deployed without any modification and
```
Comment (johannes-b) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:86]: helm -> Helm

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 3]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+applied as it is.
+
+Assuming that you've created a project from a shipyard file containing a stage with `deploymentstrategy: user_managed`,you need to:
```
Comment (johannes-b) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:89]: space between: ... user_managed`, you

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 4]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+(2) add your desired Helm chart to each stage:
+```
+keptn add-resource --project=<project-name> --service=<service-name> --all-stages --resource=<your-helm-chart.tgz> --resourceUri=helm/<service-name>.tgz
+```
+(3) send an event to trigger the delivery of your service:
+
+```
+keptn send event --file=./delivery.json 
+```
+where the content of `delvier.json` looks something like:
```
Comment (johannes-b) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:104]: delivery.json

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 5]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+  "source": "https://github.com/keptn/keptn/cli#configuration-change",
+  "specversion": "1.0",
+  "type": "sh.keptn.event.<stage-name>.delivery.triggered"
+}
+```
+
+**Limitations:**
+- Currently, when using this deployment strategy, you cannot use the `keptn trigger delivery` command as you would
+normally do. Hence, you need to send an event using `keptn send event` like described above.
+- Modifications to the helm chart via keptn configuration changes are currently not possible.
```
Comment (johannes-b) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:123]: helm --> Helm

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 6]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+
+If the deployment strategy of a stage in the [shipyard](https://github.com/keptn/examples/blob/0.8.0/onboarding-carts/shipyard.yaml)
+is configured as `deploymentstrategy: user_managed`, the provided helm chart will be deployed without any modification and
+applied as it is.
+
+Assuming that you've created a project from a shipyard file containing a stage with `deploymentstrategy: user_managed`,you need to: 
+
+(1) create a service in your stage:
+```
+keptn create service <service-name> --project <proejct-name>
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:93]: typo: proejct-name

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 7]
```diff
@@ -75,6 +75,55 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+
+```
+{
+  "contenttype": "application/json",
+  "data": {
+    "project": "<project-name",
+    "service": "<service-name>",
+    "stage": "<stage-name>" 
+  },
+  "source": "https://github.com/keptn/keptn/cli#configuration-change",
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:114]: Source should be something different. It's certainly not coming from the Keptn CLI, nor is it a configuration-change.
@johannes-b any idea?

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md (hunk 5 above)
Comment (christian-kreuzberger-dtx) [reply content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:123]: More limitations:
- Helm-Service is not able to determine the internal nor public deployment URIs. Any subsequent steps that require the URIs (e.g. jmeter tests) will not work

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md (hunk 7 above)
Comment (warber) [reply content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:114]: agree, what about `keptn-user` ? afaik the bridge does not really care (?)

Comment (christian-kreuzberger-dtx) [reply content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:114]: None of the components care, but for traceability reasons it makes sense to have something in there that shows where it request is coming from. IMHO `keptn-user` is good (or if we want to use the same wording as with quality-gates, `keptn-cli`) in there.

**Comment by christian-kreuzberger-dtx** [comment]
I see that several things have already been marked as resolved in this PR, but the actual changes are not available on GitHub yet. did you forget to commit/push?

**Comment by warber** [comment]
> I see that several things have already been marked as resolved in this PR, but the actual changes are not available on GitHub yet. did you forget to commit/push?

review comments are now incorportated

File: content/docs/0.8.x/continuous_delivery/deployment_helm/index.md [hunk 8]
```diff
@@ -75,6 +75,54 @@ carts-primary   1/1     1            1            4m   carts        docker.io/ke
...
+  "source": "keptn-cli",
+  "specversion": "1.0",
+  "type": "sh.keptn.event.<stage-name>.delivery.triggered"
+}
+```
+
+**Limitations:**
+- Currently, when using this deployment strategy, you cannot use the `keptn trigger delivery` command as you would
+normally do. Hence, you need to send an event using `keptn send event` like described above.
+- Modifications to the Helm chart via keptn configuration changes are currently not possible.
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/continuous_delivery/deployment_helm/index.md:122]: Formatting is weird in the rendered version:

image

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM - @johannes-b  you decide when and if to merge

**Pull request merged**